AlphaWallet/Settings/ViewControllers/EditCustomRPCViewController.swift

```diff
@@ -0,0 +1,155 @@
+        view = rootView
+        editView = rootView
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configure()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        keyboardChecker.viewWillAppear()
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        keyboardChecker.viewWillDisappear()
+    }
+    
+    private func configure() {
+        editView.chainNameTextField.value = viewModel.chainName
+        editView.chainNameTextField.delegate = self
+        
+        editView.rpcEndPointTextField.value = viewModel.rpcEndPoint
+        editView.rpcEndPointTextField.delegate = self
+        
+        editView.chainIDTextField.value = viewModel.chainID
+        editView.chainIDTextField.delegate = self
+        
+        editView.symbolTextField.value = viewModel.symbol
+        editView.symbolTextField.delegate = self
+        
+        editView.explorerEndpointTextField.value = viewModel.explorerEndpoint
+        editView.explorerEndpointTextField.delegate = self
+        
+        editView.configureView()
+        
+        let tap = UITapGestureRecognizer(target: self, action: #selector(tapSelected))
+        editView.addBackgroundGestureRecognizer(tap)
+        editView.isTestNetworkView.configure(viewModel: SwitchViewViewModel(text: R.string.localizable.addrpcServerIsTestnetTitle(), isOn: viewModel.isTestnet))
+        editView.configureKeyboard(keyboardChecker: keyboardChecker)
+        editView.addSaveButtonTarget(self, action: #selector(saveCustomRPC))
+        
+        navigationItem.title = R.string.localizable.editCustomRPCNavigationTitle(preferredLanguages: nil)
+    }
+    
+    @objc private func tapSelected(_ sender: UITapGestureRecognizer) {
+        view.endEditing(true)
+    }
+    
+    @objc private func saveCustomRPC(_ sender: UIButton) {
+        let result = viewModel.validate(
+            chainName: editView.chainNameTextField.value,
+            rpcEndpoint: editView.rpcEndPointTextField.value,
+            chainID: editView.chainIDTextField.value,
+            symbol: editView.symbolTextField.value,
+            explorerEndpoint: editView.explorerEndpointTextField.value,
+            isTestNet: editView.isTestNetworkView.isOn)
+        
+        switch result {
+        case .failure(.list(let errors)):
+            handleValidationFailure(errors: errors)
+        case .failure:
+            handleValidationFailure(errors: [.unknown])
+        case .success(let customRPC):
+            handleValidationSuccess(customRPC: customRPC)
+        }
+    }
+    
+    private func handleValidationFailure(errors: [EditCustomRPCErrors]) {
+        editView.resetAllTextFieldStatus()
+        for error in errors {
+            switch error {
+            case .unknown, .list:
+                // ??? TODO: Where do we log the error???
```

Comment:
Have to point this out in the PR so reviewers know to look for this.

Reply:
This code is removed due to the refactoring in [here](https://github.com/AlphaWallet/alpha-wallet-ios/pull/3417#discussion_r746240224).